CharlieBackend.Business/Services/LessonService.cs

```diff
@@ -287,5 +289,22 @@ public async Task<Result<LessonDto>> GetLessonByIdAsync(long lessonId)
 
             return Result<LessonDto>.GetSuccess(lessonDto);
         }
+
+        public async Task<bool> IsLessonDoneAsync(long id)
+        {
+            var lesson = await _unitOfWork.LessonRepository.GetByIdAsync(id);
+
+            if (lesson is null)
+            {
+                throw new NotFoundException("Given lesson not found");
+            }
+
+            if(lesson.Visits.Where(visit => visit.Presence == true).Any())
+            {
+                return true;
+            }
+
+            throw new LessonNotDoneException("The lesson was not done");
```

Comment:
Return bool value here and remove LessonNotDoneException as we don't use it for now.

Reply:
Done
![Screenshot_6](https://user-images.githubusercontent.com/56502713/112656050-ab2cb700-8e59-11eb-8b88-65673b717818.png)